randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -449,6 +492,9 @@ def _door_patches(self) -> list[dict[str, str]]:
                 print(f"Invalid door (no actor): {node}")
                 continue
 
+            if any(entry["actor"] == self._teleporter_ref_for(node) for entry in result):
```

Comment:
On a side note, I wonder how slow these `any` loops are.
I definitely will mess it up on mobile but given we change 200 doors, it will always need to run to the end of the list. For the first one there are 0 entries, for the second 1 entry...so the sum from 0 to 199 = 19900 iterations. As every door is twice in the list, but the second time it will not be added because it will be found by the loop, it runs approximately through half of the list, so additional 9950 iterations. Unless it's sorted then it will always run again to the end of the list and adding 19900 iterations.

Worst-case is like 39800 iterations and compares between two dicts.

Reply:
I only had a few seconds and did a super fast and rough test with `time.perf_counter()`: Before it took 0.436 ms and now 11.966 ms.
That is almost 30 times slower.
12 ms is not super slow, but for what it does, it's not really fast either.
Also I know this code isn't called very often and compared to how long it takes to export the game it's like nothing, but since this is something that is called on the server for multiworld (with door types rando) I wanted to at least mention it.

Reply:
Sets can't have duplicate entries right? Couldn't we just have `result` be a set and then append to that which would handle the duplicates automatically, then convert the set to a list at the end?